.github/workflows/native.yml

```diff
@@ -0,0 +1,46 @@
+# Goal here is to run all CI that do not require the ESP-IDF environment
+name: CI for Native
+
+on:
+  workflow_call:
+
+env:
+  CARGO_INCREMENTAL: 0
+  RUSTFLAGS: "-D warnings"
+  RUST_BACKTRACE: short
+  RUSTUP_MAX_RETRIES: 10
+
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+    steps:
+    - name : Checkout main branch code
+      if: github.event_name != 'pull_request_target'
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 2
+    - name: Check out PR branch code
+      if: github.event_name == 'pull_request_target'
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+        fetch-depth: 2
+    - uses: dtolnay/rust-toolchain@1.83
+      with:
+        components: clippy
+    - name: Cache Dependencies
+      uses: Swatinem/rust-cache@v2
+    - name: install libudev
+      run: sudo apt update && sudo apt install libudev-dev -y
```

Comment:
needed to build the test suite

Reply:
Maybe it belongs in the canon image definition then?

Reply:
the idea is to avoid pulling the docker image since it's big

Reply:
I'd missed that we aren't in the docker container here, this is fine.